baseapp/abci.go

```diff
@@ -252,18 +251,18 @@ func (app *BaseApp) CheckTx(ctx context.Context, req *abci.RequestCheckTx) (*abc
 
 // DeliverTxBatch executes multiple txs
 func (app *BaseApp) DeliverTxBatch(ctx sdk.Context, req sdk.DeliverTxBatchRequest) (res sdk.DeliverTxBatchResponse) {
-	scheduler := tasks.NewScheduler(app.concurrencyWorkers, app.TracingInfo, app.DeliverTx)
-	// This will basically no-op the actual prefill if the metadata for the txs is empty
-
-	// process all txs, this will also initializes the MVS if prefill estimates was disabled
-	txRes, err := scheduler.ProcessAll(ctx, req.TxEntries)
-	if err != nil {
-		// TODO: handle error
-	}
-
 	responses := make([]*sdk.DeliverTxResult, 0, len(req.TxEntries))
-	for _, tx := range txRes {
-		responses = append(responses, &sdk.DeliverTxResult{Response: tx})
+	// avoid overhead for empty batches
+	if len(req.TxEntries) > 0 {
```

Comment:
nit: slight preference for defensive checks with early return as opposed to nested ifs

Reply:
good point - that's idiomatic golang